### PR TITLE
Spark Enrich: use sbt-buildinfo (closes #3628)

### LINKED ISSUE #3625: Scala Common Enrich: use sbt-buildinfo
https://github.com/sbt/sbt-buildinfo this will remove the need for [scalifySettings](https://github.com/snowplow/snowplow/blob/master/3-enrich/scala-common-enrich/project/BuildSettings.scala#L58-L71)

### LINKED ISSUE #3628: Spark Enrich: use sbt-buildinfo
sister to #3625

### PR DESCRIPTION
**Pull request by arunma**

### CODE REVIEW

**Comment by arunma** [comment]
Seems like Travis failed with `apt-get update`.   Is there a way I could trigger the build again?
```
Using Scala 2.11.12
3.34s$ sudo apt-get -qq update
W: http://ppa.launchpad.net/couchdb/stable/ubuntu/dists/trusty/Release.gpg: Signature by key 15866BAFD9BCC4F3C1E0DFC7D69548E1C17EAB57 uses weak digest algorithm (SHA1)
W: GPG error: https://packagecloud.io/github/git-lfs/ubuntu trusty InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 6B05F25D762E3157
W: The repository 'https://packagecloud.io/github/git-lfs/ubuntu trusty InRelease' is not signed.
W: There is no public key available for the following key IDs:
6B05F25D762E3157  
E: Failed to fetch http://security.ubuntu.com/ubuntu/dists/trusty-security/main/source/Sources.gz  Hash Sum mismatch
E: Failed to fetch http://security.ubuntu.com/ubuntu/dists/trusty-security/main/binary-amd64/Packages.gz  Hash Sum mismatch
E: Some index files failed to download. They have been ignored, or old ones used instead.
The command "sudo apt-get -qq update" failed and exited with 100 during .
Your build has been stopped.
```

**Comment by chuwy** [comment]
Triggered it for you @arunma

File: 3-enrich/spark-enrich/build.sbt [hunk 1]
```diff
@@ -12,35 +12,34 @@
...
-      Dependencies.Libraries.scalaz7,
-      Dependencies.Libraries.scopt,
-      Dependencies.Libraries.commonEnrich,
-      Dependencies.Libraries.igluClient,
-      // Scala (test only)
-      Dependencies.Libraries.specs2,
-      // Thrift (test only)
-      Dependencies.Libraries.snowplowRawEvent,
-      Dependencies.Libraries.collectorPayload
-    )
```
Comment (BenFradet) [inline 3-enrich/spark-enrich/build.sbt]: could you leave all this as is? thanks.

File: 3-enrich/spark-enrich/project/BuildSettings.scala [hunk 2]
```diff
@@ -51,24 +55,14 @@ object BuildSettings {
...
-    }.taskValue
+  lazy val buildInfoSettings = Seq [BuildInfoKey](
+    organization,
+    name,
+    version,
+    "commonEnrichVersion" -> Dependencies.V.commonEnrich
   )
 
-  lazy val buildSettings = basicSettings ++ scalifySettings
+  lazy val buildInfoPackage = "com.snowplowanalytics.snowplow.enrich.spark.generated"
```
Comment (arunma) [inline 3-enrich/spark-enrich/project/BuildSettings.scala]: Please let me know if this is alright @BenFradet

File: 3-enrich/spark-enrich/build.sbt (hunk 1 above)
Comment (arunma) [reply 3-enrich/spark-enrich/build.sbt]: Absolutely !

**Comment by BenFradet** [comment]
Incorporated into #3965 , thanks a lot! :+1:

**Pull request closed without merging**